inst/database/schema.yml

```diff
@@ -0,0 +1,148 @@
+    - hash: {type: TEXT}
+    - size_csv: {type: BIGINT}
+    - size_rds: {type: BIGINT}
+
+# Every time we see a new file it'll end up here.  This would be a
+# useful starting place for deduplicating the orderly archive in a
+# fairly safe way.  I don't think that we want to (necessarily) try
+# and index all files but certainly key inputs and outputs.
+file:
+  columns:
+    - hash: {type: TEXT}
+    - size: {type: BIGINT}
+
+# Enum table for types of file use
+file_use:
+  columns:
+    - name: {type: TEXT}
+  values:
+    - {name: source}
+    - {name: script}
+    - {name: resource}
+    - {name: orderly_yml}
+    - {name: dependency}
+
+# A realisation of a file requires a filename.  These are scoped by
+# the report version and by the use within the report.
+file_input:
+  columns:
+    - id: {type: SERIAL}
+    - report_version: {fk: report_version.id}
+    - file_hash: {fk: file.hash}
+    - filename: {type: TEXT}
+    # In theory 'use' would be derivable from where the other end of
+    # the key goes but that's going to be super annoying to work out
+    - file_use: {fk: file_use.name}
+    # This would be great to have but seems overkill
+    # - mime: {type: TEXT}
+
+# Link views into the report
+report_version_view:
+  columns:
+    - id: {type: SERIAL}
+    - report_version: {fk: report_version.id}
+    - name: {type: TEXT}
+    - sql: {type: TEXT}
+
+# This is the input to get the data
+report_version_data:
+  columns:
+    - id: {type: SERIAL}
+    - report_version: {fk: report_version.id}
+    - name: {type: TEXT}
+    - sql: {type: TEXT}
+    - hash: {fk: data.hash}
+
+# What versions of packages were loaded?
+report_version_package:
+  columns:
+    - id: {type: SERIAL}
+    - report_version: {fk: report_version.id}
+    - package_name: {type: TEXT}
+    - package_version: {type: TEXT}
+
+# Values here will be populated by orderly as we keep a list in the
+# package and we should not repeat that definition.
+artefact_format:
+  columns:
+    - name: {type: TEXT}
+
+report_version_artefact:
+  columns:
+    - id: {type: SERIAL}
+    - report_version: {fk: report_version.id}
+    - format: {fk: artefact_format.name}
+    - description: {type: TEXT}
+    - order: {type: INTEGER}
+
+# This is _close_ to the same structure as file_input
+file_artefact:
+  columns:
+    - id: {type: SERIAL}
+    - artefact: {fk: report_version_artefact.id}
+    - file_hash: {fk: file.hash}
+    - filename: {type: TEXT}
+
+
+# We could go more detailed here and get the report name and id but
+# that would duplicate information so I'm not doing that here.  (The
+# information redundant in the yml/rds but that's ok I feel).
+#
+# This is a bit of a risk because it is going to assume that the
+# artefact really was included in the upstream report.  But we can
+# depend on _any_ file not just dependencies, so this might fail for
+# montagu-reports which would make this inviable.  That should be
+# tightened up a lot in the recipe_read!
+depends:
```

Comment:
We probably need a `pinned` vs `use latest` flag here to indicate that the report really, truly did want to use this version of this report and not just that this was the latest version at the time